annotated.go

```diff
@@ -159,11 +159,20 @@ var _ Annotation = paramTagsAnnotation{}
 //
 // If there has already been a ParamTag that was applied, this
 // will return an error.
+//
+// If the tag is invalid and has mismatched quotation for example,
+// (`tag_name:"tag_value') , this will return an error
 
 func (pt paramTagsAnnotation) apply(ann *annotated) error {
 	if len(ann.ParamTags) > 0 {
 		return errors.New("cannot apply more than one line of ParamTags")
 	}
+	for _, tag := range pt.tags {
+		if strings.Contains(tag, "'") {
+			correctedTag := strings.ReplaceAll(tag, "'", "\"")
+			return errors.New("the ParamTags tag has invalid quotations, did you mean " + correctedTag + " ?")
+		}
+	}
```

Comment:
Hi, just doing a quick drive-by.
This seems a bit too specific. What if my tag value has `'` in it?

```
type foo struct {
  Value string `name:"a'bc"`
```

Fx does not have any rules rejecting that, AFAIK.

Y'all might be able to borrow struct tag validation from the structtag lint check in `go vet`. 
https://cs.opensource.google/go/go/+/master:src/cmd/vendor/golang.org/x/tools/go/analysis/passes/structtag/structtag.go;drc=31881f87873b84709a49ca17195bbe5b3f683acf;l=212

(We don't need the xml/json specialization there.)

Reply:
Thanks @abhinav, that's really helpful! I'll look into that